crates/ruff_python_parser/src/lexer.rs

```diff
@@ -197,11 +202,18 @@ impl<'source> Lexer<'source> {
             _ => {}
         }
 
-        self.cursor.eat_while(is_identifier_continuation);
-
-        let text = self.token_text();
-
-        let keyword = match text {
+        // Keep track of whether the identifier is ASCII-only or not.
+        //
+        // This is important because Python applies NFKC normalization to
+        // identifiers: https://docs.python.org/3/reference/lexical_analysis.html#identifiers.
+        // We need to therefore do the same in our lexer, but applying NFKC normalization
+        // unconditionally is extremely expensive. If we know an identifier is ASCII-only,
+        // (by far the most common case), we can skip NFKC normalization of the identifier.
+        let mut is_ascii = first_char_is_ascii;
+        self.cursor
+            .eat_while(|c| is_identifier_continuation(c, &mut is_ascii));
+
+        let keyword = match self.token_text() {
```

Comment:
Nit: We can skip the entire `match` for `non-ascii` text. The old code didn't do this to avoid branching but we might as well do it now, where we branch on the flag anyway.